### ISSUE TITLE
BadFormatException from current pull as of 21st of March, 2010

### ISSUE THREAD

**Issue by jacobkjaer** [headsling/Node.cs#1]
Hey, I just stumbled upon this project, and it looks interesting. I pulled the latest release onto my machine, opened it in VS2010, and compiled it.

Now running node.exe, with or without example.*.dll as an argument, produces this exception:

`Unhandled Exception: System.BadImageFormatException: Could not load file or assembly 'Manos, Version=0.1.3.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies. An attempt was made to load a program with an incorrect format.  
   at node.Program.Main(String[] args)`

I have tried to remove the /bin folder before compilation to make sure everything is up to date, but to no avail. Help would be appreciated, as there seem to be no guidelines stating that I am going something wrong.

**Comment by headsling** [comment]
Hey - sorry about that!  I'm developing on linux and am not really tracking the windows side of things.  I'd probably start by looking at getting manos de mono, which i'm currently relying on for my event loop (http://groups.google.com/group/manos-de-mono/browse_thread/thread/f9e84d4221968742 is a starting point).  I'll see if I can make some time this weekend to have a look.  

Out of interest - are you running 64bit or 32?

**Comment by jacobkjaer** [comment]
Ah, of course, you're on a Mono system, that's why :) I'm on a 64-bit Win7 installation myself. I don't know if I'll look further into getting this to compile, but feel free to poke me if you want me to test anything in the future.

**Comment by headsling** [comment]
can you try and force your compile to x86?  i'm pretty sure that will get you past the badformat error, but then we'll be on to the libev library issues.

**Comment by jacobkjaer** [comment]
Yup, that worked perfectly. I'm getting libev errors now, but that's probably because it hasn't been installed on my machine yet (duh), I'll do it and keep you updated.

**Comment by jacobkjaer** [comment]
Yup, libev troubles. I managed to compile libev.dll, but I am sure the config.h file I used was wrong. I'm getting:
`Unhandled Exception: System.TypeInitializationException: The type initializer for 'Manos.IO.IOLoop' threw an exception. ---> System.Exception: No supported back
end in libev`

I'll look it up online and see how I can get it to play along with me.

**Comment by jacobkjaer** [comment]
(Oh, and sorry, didn't see your Google groups link at first, it's still a bit early for me ;))

**Comment by headsling** [comment]
hey - thanks for looking into this! did you get any further?

**Comment by jacobkjaer** [comment]
A little! Here's what I did:
- compiled Node.cs as x86
- downloaded a Manos de Mono source-tree which is supposed to run on Windows according to the author on Google Groups
- compiled libev.dll from that MdM source-tree
- put the new libev.dll and manos libraries in the same folder as my node.exe

Here is the result: http://www.student.dtu.dk/~s072421/i/libeverror.png

I don't know if you will be able to re-produce this. But as of now, I seem to have been moved onto another project where I won't been working with Node.cs at first hand. I hope however, just out of personal interest, to see a mature Windows support in the future. Good luck :)
